homeassistant/components/ssdp/__init__.py

```diff
@@ -425,11 +428,26 @@ async def async_start(self) -> None:
 
     async def _async_start_ssdp_listeners(self) -> None:
         """Start the SSDP Listeners."""
+        has_scope_id = sys.version_info >= (
+            3,
+            9,
+        )
         for source_ip in await self._async_build_source_set():
```

Comment:
It looks like `_async_build_source_set` will only give IPv4 addresses if the user hasn't enabled any non-default adapters. This is not ideal, as it stops SSDP listening on IPv6 addresses altogether.